### DISCUSSION TITLE
Not Connected To MQTT Server!

### DISCUSSION THREAD

**Discussion by NimKat16** [Koenkk/zigbee2mqtt#9180] [General]
After  spending many hours, i finally got my CC2652R recognized by zigbee2mqtt and now i can even see the one device I have sending messages which are logged. Nice! BUT I cannot connect to MQTT broker! :(
I have installed Mosquitto, used the integration page to enable and configure it, added a new user which is not admin. Added the user to zigbee2mqtt config, restarted zigbee2mqtt, mosquitto, HA, the whole PC a few times.. but no difference! I get below message

![image](https://user-images.githubusercontent.com/91806336/137601071-e236453b-5004-495a-8b80-f6d6776c4412.png)

I have spend 3 days searching over for the solution but i cannot find any answer... i wonder if my server configuration is correct. It is now server: mqtt://homeassistant.local:1883, should it not be http:// homeassistant.local:1883?

The logs of Mosquitto do not show any thing! I added log_types: all but the only thing i see is shown below

![image](https://user-images.githubusercontent.com/91806336/137601193-cc51d21f-d3ff-4d70-a667-bc9f9cf72805.png)

My Mosquitto config is shown below

![image](https://user-images.githubusercontent.com/91806336/137601235-92e546f8-cf2a-41b5-9be8-394a443730a5.png)

I guess i have missed an step or have not activated something but i cannot figure it out.. What is that I am missing?

**Comment by NimKat16** [comment]
Now this is getting more difficult... I updated HA core and Operating system and the addons restarted. Now I see a few additional line in Mosquitto log

![image](https://user-images.githubusercontent.com/91806336/137601564-d4f023e4-f83a-4466-88ac-9e083ec21d8e.png)

Why there is a socket error and disconneting?
There are so many socket error issues reported but none that i have read seems to be relevant to this one... Mosquitto log hangs there and nothing is added to it!

**Comment by francisp2** [comment]
I only use

```
mqtt:
  base_topic: zigbee2mqtt
  server: mqtt://10.0.0.63
  user: xxx
  password: xxx
```

**Reply by NimKat16** [reply]
Thanks Francisp2 for your reply :)
How did you get this 10.0.0.63? I looked everywhere to find the internal IP of the broker in vain.. or I am blind! :(

I tried to use the test server putting server: http://test.mosquitto.org:1883 in zigbee2mqtt and it works!...

**Comment by NimKat16** [comment]
Now I should really thank you again Francisp2. Well, kudos to you, because you definitely solved the 2 problems I had. After I wrote above reply i thought of trying one more time the Host IPv4 address found on Supervisor > System > Host . In the past I did try it in the form of mqtt://<IP>:1883 and http://<IP>:1883 and both had failed...

Now I only used the in the form of mqtt://<IP> with no port. zigbee2mqtt does an auto discovery and finds it...

I wished the documentation could be corrected for this. That makes the life of newbies easier.. :)
